### PR TITLE
More compose

### PR DESCRIPTION
**Pull request by athoune**

### CODE REVIEW

**Comment by athoune** [comment]
Network subnet are now explicitly handled. `ComposeValidator` and `Recomposator` patterns.

File: compose/network.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+}
+
+func (s Subnet) Next() (Subnet, error) {
+	r := Subnet{}
+	if s[1] < 255 {
+		r[1] = s[1] + 1
+		r[0] = s[0]
+	} else {
+		if s[0] > 31 {
+			return r, errors.New("Too large")
```
Comment (papey) [inline compose/network.go:40]: `fmt.Errorf` > `errors.New`

**Comment by papey** [comment]
`--- FAIL: TestNetworkNew (0.00s)`

Fails on CI, ok my machine, maybe remove this test when running inside CI.

**Comment by papey** [comment]
LGTM.

**Pull request merged**